### PR TITLE
allow forum delivery using attached mention tags

### PR DESCRIPTION
**Pull request by redmatrix**
Compatibility: This is completely untested as I do not have an installed friendica site to test against; but should allow forum deliveries if mention is contained in attached tags besides only pattern matching in the message body.

### CODE REVIEW

File: src/Model/Item.php [hunk 1]
```diff
@@ -2670,6 +2671,16 @@ private static function tagDeliver($uid, $item_id)
 			}
 		}
 
+		if (!$mention) {
+			$tags = Tag::getByURIId($item_id, [Tag::MENTION, Tag::EXCLUSIVE_MENTION]);
+			foreach ($tags as $tag) {
+				if (Strings::compareLink($link, $tag['url']) || Strings::compareLink($dlink, $tag['url'])) {
+					$mention = true;
+					Logger::log('mention found in tag: ' . $tag['url']);
```
Comment (nupplaphil) [inline src/Model/Item.php]:
```suggestion
					DI::logger()->info('mention found in tag.', ['url' => $tag['url']]);
```

Comment (redmatrix) [reply src/Model/Item.php]: I'm not able to determine the correct way to log the event and copied the previous few lines where the same mechanism was used. My apologies - could you please adjust as required?

**Review by MrPetovan — APPROVED** [review]

File: src/Model/Item.php [hunk 2]
```diff
@@ -2670,6 +2671,16 @@ private static function tagDeliver($uid, $item_id)
 			}
 		}
 
+		if (!$mention) {
+			$tags = Tag::getByURIId($item_id, [Tag::MENTION, Tag::EXCLUSIVE_MENTION]);
```
Comment (MrPetovan) [inline src/Model/Item.php]: `$item_id` doesn't contain the `uri-id` expected by `Tag::getByURIId`, I believe it should be something like
```suggestion
			$tags = Tag::getByURIId($item['uri-id'], [Tag::MENTION, Tag::EXCLUSIVE_MENTION]);
```

**Review by MrPetovan — CHANGES_REQUESTED** [review]

File: src/Model/Item.php (hunk 2 above)
Comment (annando) [reply src/Model/Item.php]: Yes, it should.

**Review by MrPetovan — APPROVED** [review]

**Pull request merged**